pkg/skaffold/docker/client.go

```diff
@@ -39,18 +41,39 @@ type DockerAPIClient interface {
 	io.Closer
 }
 
+var (
+	dockerAPIClientOnce sync.Once
+	dockerAPIClient     DockerAPIClient
+	dockerAPIClientErr  error
+)
+
 // NewDockerAPIClient guesses the docker client to use based on current kubernetes context.
-func NewDockerAPIClient(kubeContext string) (DockerAPIClient, error) {
+func NewDockerAPIClient() (DockerAPIClient, error) {
+	dockerAPIClientOnce.Do(func() {
```

Comment:
do both of these have to be wrapped in sync.Once? Only either the NewDockerAPIClient or CurrentContext should need to be memoized. I think just the NewDockerAPIClient would be sufficient, since current context should be quick enough (it just parses the kubeconfig and doesn't talk to the server afaik) whereas the new docker API client getter actually pings the docker endpoint